### PR TITLE
Recognition error handling

### PR DESCRIPTION
**Pull request by domanttas**
@tomasmuzas So, I tried to change error handling in recognition API as we have discussed. I think it's pretty solid now. If you give it a "gogo", I'll change error handling in whole project in the same way.

NOTE: I left exclamation mark in error message so frontend wouldn't blow up, it's a 'TODO' to remove it 
    and translate messages in some way.

### CODE REVIEW

File: WhoIsThatServer.Recognition/Recognition/RecognitionServices.cs [hunk 1]
```diff
@@ -76,30 +68,27 @@ public async Task<bool> CreateGroup()
...
+                throw new IndexOutOfRangeException();
 
             var faceIds = faces.Select(face => face.FaceId).ToArray();
 
             var results = await _faceServiceClient.IdentifyAsync(_groupId, faceIds);
-            
+
             if (results.Length == 0 || results == null || results[0].Candidates.Length == 0 || results[0].Candidates[0] == null)
-                return "No one was indetified!";
+                throw new ArgumentNullException();
```
Comment (tomasmuzas) [inline WhoIsThatServer.Recognition/Recognition/RecognitionServices.cs]: It's getting closer, but not yet ;) how would you differentiate between empty uri and unrecognized face?

Comment (tomasmuzas) [reply WhoIsThatServer.Recognition/Recognition/RecognitionServices.cs]: Would be nice, if you added error code to the exception

Comment (domanttas) [reply WhoIsThatServer.Recognition/Recognition/RecognitionServices.cs]: I thought about that as well.

By error code, do you mean to add some login to catch block in controller, for example:

catch(RandomException ex) when (ex.Status == someStatus)

Or getting information from exception in some other way, for example creating custom exception "EmptyUrlException"?

Comment (tomasmuzas) [reply WhoIsThatServer.Recognition/Recognition/RecognitionServices.cs]: Yeah, I meant catch/when

Comment (tomasmuzas) [reply WhoIsThatServer.Recognition/Recognition/RecognitionServices.cs]: And you could also have a general `ManagerException` or something like this

Comment (domanttas) [reply WhoIsThatServer.Recognition/Recognition/RecognitionServices.cs]: Okay, cool, I'll modify it with catch/when, thanks!

I understand the concept of having general ManagerException, but can't wrap my head around on how to implement it, but maybe I'm just tired today.

Comment (tomasmuzas) [reply WhoIsThatServer.Recognition/Recognition/RecognitionServices.cs]: Something as:
```
....
throw new ManagerException(ErrorCode.FaceNotDetected);

throw new ManagerException(ErrorCode.WrongUri);

...
catch(ManagerException ex) when (ex.ErrorCode == ErrorCode.FaceNotDetected)
```

**Comment by domanttas** [comment]
@tomasmuzas I did solve this by adding message to exception, but still doesn't feel right. I guess it was the easiest solution on the spot. Though, is it okay to separate exceptions by messages, not status?

File: WhoIsThatServer.Recognition/Controllers/RecognitionServicesController.cs [hunk 2]
```diff
@@ -31,10 +31,15 @@ public async Task<JsonResult> InitiateRecognition()
                 return Json(RecognitionErrorMessages.NoFacesFoundError);
             }
 
-            catch (ArgumentNullException argumentNullException)
+            catch (ArgumentNullException imageArgumentNullException) when (imageArgumentNullException.Message.Contains("image"))
```
Comment (tomasmuzas) [inline WhoIsThatServer.Recognition/Controllers/RecognitionServicesController.cs]: Another class (for messages) and == would be much cleaner but that may do for now

File: WhoIsThatServer.Recognition/Recognition/RecognitionServices.cs (hunk 1 above)
Comment (domanttas) [reply WhoIsThatServer.Recognition/Recognition/RecognitionServices.cs]: that looks really clean, will certainly do that, thank you!!

File: WhoIsThatServer.Recognition/Controllers/RecognitionServicesController.cs (hunk 2 above)
Comment (domanttas) [reply WhoIsThatServer.Recognition/Controllers/RecognitionServicesController.cs]: Alright, anyhow, will change it to your suggestion with manager exception, it looks very good

**Comment by domanttas** [comment]
@tomasmuzas so, I implemented your suggestion with exceptions as you guided me, could you check it out if everything's fine and if I can move forward with that approach in other parts of project?

**Review by tomasmuzas — APPROVED** [review]
Looks awesome now!

**Comment by domanttas** [comment]
Thanks!!

**Pull request merged**